docker/docker-entrypoint.sh

```diff
@@ -1,16 +1,5 @@
 #!/bin/sh
 
-# Create writable dirs and change ownership to www-data
-for path in " " users products product_images orgs new_images logs; do
-  mkdir -p /mnt/podata/${path}
-  chown www-data:www-data /mnt/podata/${path}
-done
-
-# Create symlinks of data files to /mnt/podata
-for path in ecoscore emb_codes forest-footprint ingredients lang packager-codes po taxonomies templates; do
-  ln -sfT /opt/product-opener/${path} /mnt/podata/${path}
-done
-
```

Comment:
In some cases I find it okay for the entrypoint script to contain `chown` statements and even `usermod -uid $USER_UID` to allow changing UID/GID dynamically and not have to worry about at runtime.

Looking at how much moving those few lines from `docker-entrypoint.sh` to Dockerfile complexified the setup workflow, I'm not so sure if that was the right call:

* Dockerfile is now way more complex than it used to (I'm confused by it not being an absolute expert in Docker but I haven't seen `COPY --chown` commands in other Dockerfiles, as well as 5/6 lines of inline script containing `mkdir`, `chown`, `usermod` commands)
* We added an `init_backend` command that needs to be run before `up`

Reply:
For me the problem with previous setup, was that starting any task with docker was taking some 15s, which I find a bit too much if I can displace it in the Image creation (I really use a lot run --rm rather than exec and this is a good practice for say running test, or running a command whithout needing to start the apache service and it's 6G load). More over it is also better for the image to really contain all target structure, for me this is like a documentation of full structure.

I think the complexity here is brought by the lack of separation between code, configuration and data in product opener. I hope we will solve that slowly by changing the code.

For me entrypoint can transform files etc., but normally for configuration purpose, not finishing the install.

Reply:
+1 for the lack of separation between code / configuration / data --> the ideal state would be to have a stateless ProductOpener in Docker but we're far away from that :) as long as it's not too tedious for the end user and that maintainers understand what's in there that's fine for me.